### PR TITLE
Add YouTube video picker

### LINKED ISSUE #2: 📽️ YouTube video picker
Content editor can search and pick a video from YouTube. It allows him to link related YouTube video with the given content entry without actually leaving Kentico Cloud admin UI. Developer can then render selected video as a part of the output presentation to provide complete information to the content consumer.

### Accept criteria
- There is a preview of the selected video. It helps content editor to be more confident that the chosen video is the right one.

### Proposed solution
A custom element allowing to search and select video from YouTube

### PR DESCRIPTION
**Pull request by Bassetts**
Resolves #2, allows editors to search for and select YouTube videos. This was a joint effort with @ileshmistry 😄

### CODE REVIEW

File: YouTube/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# YouTube video selector
```
Comment (Simply007) [inline YouTube/README.md]: New line to follow https://github.com/DavidAnson/markdownlint/blob/v0.13.0/doc/Rules.md#md022
```suggestion
# YouTube video selector

```

File: YouTube/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+YouTube video selector allows users to search and select videos from YouTube.
+
+![YouTube video selector](YouTubeVideoSelector.gif)
+
+# Usage
```
Comment (Simply007) [inline YouTube/README.md]: Second level of heading to follow https://github.com/DavidAnson/markdownlint/blob/v0.13.0/doc/Rules.md#md025
```suggestion
## Usage
```

File: YouTube/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+To use the YouTube video selector in your Kentico Cloud project:
+
+* In Kentico Cloud open Content types tab
+* Open / create a content model to which you want to add the YouTube video selector
+* Add **Custom element** content element
+* Open configuration of the content element
+* Use following URL as Hosted code URL (HTTPS): https://kentico.github.io/custom-element-samples/YouTube/index.html
+* Provide the following JSON parameters for the custom element, to get an API key for the YouTube API vist https://developers.google.com/youtube/v3/getting-started
```
Comment (Simply007) [inline YouTube/README.md]:
```suggestion
* Provide the following JSON parameters for the custom element,
 to get an API key for the YouTube API visit https://developers.google.com/youtube/v3/getting-started
```

File: YouTube/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+```
```
Comment (Simply007) [inline YouTube/README.md]: add code definition for code block

```suggestion
```json
```

File: YouTube/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+{
+  "apiKey": "<YOUR API KEY>"
+}
+```
+
+# Installation
```
Comment (Simply007) [inline YouTube/README.md]: Second level of heading to follow https://github.com/DavidAnson/markdownlint/blob/v0.13.0/doc/Rules.md#md025

```suggestion
## Installation
```

File: YouTube/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+YoutTube video selector source code is in following repository: https://github.com/Bassetts/kc-youtube
```
Comment (Simply007) [inline YouTube/README.md]: typo 
```suggestion
YouTube video selector source code is in following repository: https://github.com/Bassetts/kc-youtube
```

File: YouTube/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+If you want to customise the element please follow the instructions in source code repository README.md
```
Comment (Simply007) [inline YouTube/README.md]: Typo
```suggestion
If you want to customize the element please follow the instructions in [source code repository `README.md`](https://github.com/Bassetts/kc-youtube/Readme.mhttps://github.com/Bassetts/kc-youtube/blob/master/README.md).
```

**Comment by Bassetts** [comment]
@Simply007 I have made the requested changes and also slightly modified the value saved to Kentico Cloud to add the video title. This was needed so the input had the correct value on the initial load.

A sample of the value saved is

```
{
  "item": {
    ...
    "elements": {
      ...
      "youtube": {
        "type": "custom",
        "name": "Youtube",
        "value": {"title":"What Is Headless CMS?","videoId":"U5_H9cD17gA"} // or null
      }
      ...
    }
  }
  ...
}
```

Did you want this adding to the README?

**Comment by Simply007** [comment]
Hello @Bassetts,
sorry for the delay. It would be great to add the sample value output to the readme. 
Thanks!

**Comment by Bassetts** [comment]
@Simply007 No worries, I have added this to the README 😄

**Comment by Simply007** [comment]
@Bassetts,

that was really quick!

I have reviewed all the changes and it looks cool. There is just one review note that is not incorporated.

> Would it be possible to display some error message if the configuration is not provided/API key is incorrect?

Something like:
image

```html
<span class="item-status item-status--failed">
  There is no, or incorrect input configuration specified for this component. Please check the documentation.
</span>
```

**Comment by Bassetts** [comment]
@Simply007 I had implemented this but had the check slightly wrong, so if the config was completely missing then the element did not load. It should be working as intended now 😄

**Comment by Simply007** [comment]
Great, there is still no notification to the user when the user has an invalid/revoked/expired key.

My configuration
```json
{
    "apiKey": 1
}
```

File: YouTube/README.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+{
+  "apiKey": "<YOUR API KEY>"
+}
+```
+
+## Installation
+
+YouTube video selector source code is in following repository: https://github.com/Bassetts/kc-youtube
+
+If you want to customize the element please follow the instructions in [source code repository `README.md`](https://github.com/Bassetts/kc-youtube/Readme.mhttps://github.com/Bassetts/kc-youtube/blob/master/README.md).
```
Comment (Simply007) [inline YouTube/README.md]:
```suggestion
If you want to customize the element please follow the instructions in [source code repository `README.md`](https://github.com/Bassetts/kc-youtube/Readme.md).
```

**Comment by Bassetts** [comment]
@Simply007 an error message is now shown when the API call returns an error message.

**Pull request merged**